lib/reversetunnel/localsite.go

```diff
@@ -523,7 +524,70 @@ func stringOrEmpty(addr net.Addr) string {
 	return addr.String()
 }
 
+func (s *localSite) setupTunnelForOpenSSHEICENode(ctx context.Context, targetServer types.Server) (*awsoidc.OpenTunnelEC2Response, error) {
+	awsInfo := targetServer.GetAWSInfo()
+	if awsInfo == nil {
+		return nil, trace.BadParameter("missing aws cloud metadata")
+	}
+
+	issuer, err := awsoidc.IssuerForCluster(ctx, s.accessPoint)
```

Comment:
Why do we need to iterate over all proxies here? Can't we get proxy public address from config or something?

Reply:
In a previous commit we had the proxy public address.
However, that changed things in a lot of places (including several tests set up).
But then, we agreed that would be better to just get the proxy address here
You can read more about it here https://github.com/gravitational/teleport/pull/30632#discussion_r1299986165

Now that I think about it, I will probably create a PR where the issuer should not be necessary, because we could get it within the Auth Service.
The problem is that, if we have multiple public addresses, we might not send the right one to AWS 🤔 
I will think about it a little more, but best scenario: we remove the issuer from here and from a couple more places.